server/api/store.go

```diff
@@ -66,6 +68,15 @@ const (
 	downStateName    = "Down"
 )
 
+// parseIDUintRespondErr gets the id from the mux.Vars or returns a 400
+func parseIDUintRespondErr(vars map[string]string, rd *render.Render, w http.ResponseWriter) (uint64, error) {
```

Comment:
em, I am surprised why responding error but still return the error, why not provide a simple `parseID` function and respond the error if error outside. 

I think this is more clear.

Reply:
I agree that this kind of abstraction is not working well in go lang where there is no way to create an early return in a handler from a called function.